lib/routes/xiaohongshu/notes.ts

```diff
@@ -15,25 +18,125 @@ export const route: Route = {
     const userId = ctx.req.param('user_id');
     const url = `https://www.xiaohongshu.com/user/profile/${userId}`;
 
-    const { user, notes } = await getNotes(url, cache);
+    if (config.xiaohongshu.cookie && ctx.req.param('fulltext')) {
+        const user = await getUser(url, config.xiaohongshu.cookie);
+        const notes = await renderNotesFulltext(user.notes, url);
+        return {
+            title: `${user.userPageData.basicInfo.nickname} - 笔记 • 小红书 / RED`,
+            description: user.userPageData.basicInfo.desc,
+            image: user.userPageData.basicInfo.imageb || user.userPageData.basicInfo.images,
+            link: url,
+            item: notes,
+        };
+    } else {
+        const { user, notes } = await getNotes(url, cache);
+        return {
+            title: `${user.nickname} - 笔记 • 小红书 / RED`,
+            description: formatText(user.desc),
+            image: user.imageb || user.images,
+            link: url,
+            item: notes.map((item) => formatNote(url, item)),
+        };
+    }
+}
+
+async function getUser(url, cookie) {
+    const res = await got(url, {
+        headers: {
+            'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.0.0 Safari/537.36',
+            Cookie: cookie,
+        },
+    });
+    const $ = cheerio.load(res.data);
+
+    let script = $('script')
+        .filter((i, script) => {
+            const text = script.children[0]?.data;
+            return text?.startsWith('window.__INITIAL_STATE__=');
+        })
+        .text();
+    script = script.slice('window.__INITIAL_STATE__='.length);
+    script = script.replaceAll('undefined', 'null');
+    const state = JSON.parse(script);
+    return state.user;
+}
+
+async function renderNotesFulltext(notes, url) {
+    const data: any[] = [];
+    const promises = notes.flatMap((note) =>
+        note.map(async ({ noteCard }) => {
+            const link = `${url}/${noteCard.noteId}`;
+            const { title, description, pubDate } = await getFullNote(link);
+            return {
+                title,
+                link,
+                description,
+                author: noteCard.user.nickName,
+                guid: noteCard.noteId,
+                pubDate,
+            };
+        })
+    );
+    data.push(...(await Promise.all(promises)));
+    return data;
+}
 
-    return {
-        title: `${user.nickname} - 笔记 • 小红书 / RED`,
-        description: formatText(user.desc),
-        image: user.imageb || user.images,
-        link: url,
-        item: notes.map((item) => formatNote(url, item)),
-    };
+async function getFullNote(link) {
+    const cookie = config.xiaohongshu.cookie;
+    const data = (await cache.tryGet(link, async () => {
+        const res = await got(link, {
+            headers: {
+                'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.0.0 Safari/537.36',
```

Comment:
RSSHub will use a randomised user agent of [Chrome on mac](https://github.com/DIYgod/RSSHub/blob/a41cfc79ec93186793bf5b98a1c46cbd9a03edd3/lib/config.ts#L440) by default. Does the site only works with this fixed version of Chrome on Windows?